.github/workflows/python-ci.yml

```diff
@@ -16,20 +16,53 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
+
     - name: Install system dependencies
       run: |
         sudo apt-get update
         sudo apt purge postgresql-client-13 postgresql-server-dev-all
-        sudo apt-get install postgresql-plpython3-12 postgresql-12-postgis-3 libhdf5-dev libgdal-dev libnetcdf-dev
-    - name: Install python dependencies
+        sudo apt-get install postgresql-plpython3-12 postgresql-12-postgis-3
+        sudo apt-get install libhdf5-dev libgdal-dev libnetcdf-dev
+
+    - name: Install pipenv
+      run: |
+        pip install -U pipenv
+
+    - id: cache-pipenv
+      uses: actions/cache@v2
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+
+    - name: Install dependencies if changed
+      if: steps.cache-pipenv.outputs.cache-hit != 'true' && ${{ matrix.python-version == '3.8' }}
+      run: |
+        pipenv install --dev
+        pipenv install -e .
+        pipenv uninstall numpy GDAL
+        pipenv run pip install numpy
+        pipenv run pip install GDAL==$(gdal-config --version) --global-option=build_ext --global-option="-I/usr/include/gdal"
```

Comment:
Well, this is hideous. Good job getting it to work.

Is there somewhere to put a comment explaining why it has to be done this way, for whoever the next person to work on it is?